src/typed_ast.cpp

```diff
@@ -171,6 +171,15 @@ TypedAST* convertAST(AST::ForStatement* ast) {
     return typed_for;
 }
 
+TypedAST* convertAST(AST::WhileStatement* ast) {
+    auto typed_for = new WhileStatement;
+
+    typed_for->m_condition = get_unique(ast->m_condition);
+    typed_for->m_body      = get_unique(ast->m_body);
+
+    return typed_for;
```

Comment:
it's a bit ugly that the variable is named `typed_for`. either `typed_while` or `typed_ast` would be more appropriate

Reply:
Woops, missed that one